pqarrow/parquet.go

```diff
@@ -150,12 +150,13 @@ func RecordToDynamicRow(schema *dynparquet.Schema, record arrow.Record, index in
 		return nil, io.EOF
 	}
 
-	ps, err := schema.DynamicParquetSchema(RecordDynamicCols(record))
+	ps, err := schema.GetDynamicParquetSchema(RecordDynamicCols(record))
```

Comment:
I think we can improve this further by pulling the dynamic schema out of the `RecordToDynamicRow` function since it calls that function for each row in the record, as well as the `RecordToDynamicSchema` function

Reply:
Done. RFAL.